plugins/woocommerce-admin/client/products/sections/utils.ts

```diff
@@ -108,3 +108,13 @@ export const getInputControlProps = ( {
 		help,
 	};
 };
+
+/**
+ * Test if a string is a valid slug.
+ *
+ * @param  string String to test
+ * @return boolean
+ */
+export const isValidSlug = ( string: string ) => {
```

Comment:
Shouldn't this also include a length check? It isn't really a valid slug if it is greater than 255 characters (core validation).

Reply:
I think you're right.  Interestingly, `@wordpress/url`'s `cleanSlug` does not take into account length, but that may be by design while validation is meant to be separate.

I'll add the length logic here if we end up keeping the validation.

Reply:
Yeah, the concepts of cleaning/sanitization and validation are overlapping but different. When talking about whether something is "valid", I think we are talking about validation, which I think is a superset that includes cleaning/sanitization and additional rules.